search-service/src/main/kotlin/com/egm/stellio/search/service/AttributeInstanceService.kt

```diff
@@ -47,7 +47,7 @@ class AttributeInstanceService(
     fun search(
         temporalQuery: TemporalQuery,
         temporalEntityAttribute: TemporalEntityAttribute
-    ): Mono<List<Map<String, Any>>> {
+    ): Mono<List<Map<String, Any?>>> {
```

Comment:
It's getting hard to read.

Add a specific data class that will contain observedAt, value, ... and map each individual result it (so that you finally return a `List<AttributeInstanceResult>` - or a better name if you find one)